### PR TITLE
fix: build break with ENABLE_LOG=OFF

### PR DESCRIPTION
**Pull request by shoamano83**
Fixes https://github.com/smartdevicelink/sdl_core/issues/1788

### CODE REVIEW

**Comment by shoamano83** [comment]
@JackLivio @jacobkeeler Please kindly review.

File: src/components/application_manager/src/policies/policy_handler.cc [hunk 1]
```diff
@@ -808,7 +808,9 @@ bool PolicyHandler::IsAppSuitableForPolicyUpdate(
     const Applications::value_type value) const {
   LOG4CXX_AUTO_TRACE(logger_);
 
+#ifdef ENABLE_LOG
   const uint32_t app_id = value->app_id();
+#endif
```
Comment (jacobkeeler) [inline src/components/application_manager/src/policies/policy_handler.cc]: Probably better just to remove this variable than to use a macro

Comment (shoamano83) [reply src/components/application_manager/src/policies/policy_handler.cc]: OK, removed by https://github.com/smartdevicelink/sdl_core/pull/1789/commits/1b8e82a38e879a022647e6e17498c9865d7853e3.

**Comment by JackLivio** [comment]
Changed to 4.4.0 target

**Comment by JackLivio** [comment]
reverted because it was giving me a compilation error when merged into 4.4.0

**Pull request merged**